src/metabase/models/dashboard_card.clj

```diff
@@ -201,48 +205,45 @@
-  {:dashboard_id                            su/IntGreaterThanZero
-   (s/optional-key :card_id)                (s/maybe su/IntGreaterThanZero)
-   (s/optional-key :action_id)              (s/maybe su/IntGreaterThanZero)
+  ;; TODO - make the rest of the options explicit instead of just allowing whatever for other keys
+  [:map
+   [:dashboard_id                            ms/PositiveInt]
+   [:action_id              {:optional true} [:maybe ms/PositiveInt]]
    ;; TODO - use ParamMapping. Breaks too many tests right now tho
-   (s/optional-key :parameter_mappings)     (s/maybe [#_ParamMapping su/Map])
-   (s/optional-key :visualization_settings) (s/maybe su/Map)
-   ;; TODO - make the rest of the options explicit instead of just allowing whatever for other keys
-   s/Keyword                                s/Any})
+   [:parameter_mappings     {:optional true} [:maybe [:sequential map?]]]
+   [:visualization_settings {:optional true} [:maybe map?]]
+   [:series                 {:optional true} [:maybe [:sequential ms/PositiveInt]]]])
 
-(s/defn create-dashboard-card!
+(mu/defn create-dashboard-cards!
   "Create a new DashboardCard by inserting it into the database along with all associated pieces of data such as
-   DashboardCardSeries. Returns the newly created DashboardCard or throws an Exception."
-  [dashboard-card :- NewDashboardCard]
-  (let [{:keys [dashboard_id card_id action_id parameter_mappings visualization_settings size_x size_y row col series]
-         :or   {series []}} dashboard-card]
-    (t2/with-transaction [_conn]
-      (let [dashboard-card (first (t2/insert-returning-instances!
-                                    DashboardCard
-                                    :dashboard_id           dashboard_id
-                                    :card_id                card_id
-                                    :action_id              action_id
-                                    :size_x                 size_x
-                                    :size_y                 size_y
-                                    :row                    row
-                                    :col                    col
-                                    :parameter_mappings     (or parameter_mappings [])
-                                    :visualization_settings (or visualization_settings {})))]
-       ;; add series to the DashboardCard
-       (update-dashboard-card-series! dashboard-card series)
-       ;; return the full DashboardCard
-       (retrieve-dashboard-card (:id dashboard-card))))))
-
-(defn delete-dashboard-card!
-  "Delete a DashboardCard."
-  [dashboard-card user-id]
-  {:pre [(map? dashboard-card)
-         (integer? user-id)]}
-  (let [{:keys [id]} (dashboard dashboard-card)]
+  DashboardCardSeries. Returns the newly created DashboardCard or throws an Exception."
+  [dashboard-cards :- [:sequential NewDashboardCard]]
+  (t2/with-transaction [_conn]
+    (let [dashboard-card-ids (t2/insert-returning-pks!
```

Comment:
We usually check if the input is empty, but not in this case. Is that intentional?

Reply:
not intentional, I'll add a seq check.

Reply:
0d7da010b2